### PR TITLE
Validate interface name passed in ECS_OFFHOST_INTROSPECTION_INTERFACE_NAME

### PR DESCRIPTION
**Pull request by sparrc**
### Summary

ECS_OFFHOST_INTROSPECTION_INTERFACE_NAME allows system administrators to input an interface name in the ecs.config file. This user input is then passed into an `exec` iptables command here: https://github.com/aws/amazon-ecs-agent/blob/4480f9d7d788dd9e9100df3b685d173bdd4b41c2/ecs-init/exec/iptables/iptables.go#L206-L222

ecs.config is a protected file that only the root user has write access to, so the threat is limited. But we should still validate that this is a valid linux interface name anyways, so that this env var cannot be used to inject an unexpected command into the middle of this iptables command.

### Testing

New tests cover the changes: yes

### Description for the changelog

Enhancement: Validate ECS_OFFHOST_INTROSPECTION_INTERFACE_NAME is a valid interface name.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-init/exec/iptables/iptables.go [hunk 1]
```diff
@@ -213,10 +214,38 @@ func getBlockIntrospectionOffhostAccessInputChainArgs() []string {
...
+// https://github.com/torvalds/linux/blob/d7e78951a8b8b53e4d52c689d927a6887e6cfadf/net/core/dev.c#L1056-L1079
+func checkValidIfname(name string) error {
+	if name == "" {
+		return fmt.Errorf("Invalid ifname [%s]: empty string not allowed", name)
+	}
+	if len(name) >= 16 {
+		return fmt.Errorf("Invalid ifname [%s]: length of name must be less than 16 chars", name)
+	}
+	if name == "." || name == ".." {
+		return fmt.Errorf("Invalid ifname [%s]: name or '.' or '..' not allowed", name)
```
Comment (singholt) [inline ecs-init/exec/iptables/iptables.go:229]: return fmt.Errorf("Invalid ifname [%s]: ~~name or~~ '.' or '..' not allowed", name)

File: ecs-init/exec/iptables/iptables.go [hunk 2]
```diff
@@ -213,10 +214,38 @@ func getBlockIntrospectionOffhostAccessInputChainArgs() []string {
...
+	}
+
+	for _, char := range name {
+		if char == '/' || char == ':' || unicode.IsSpace(char) {
+			return fmt.Errorf("Invalid ifname [%s]: Space characters, ':', and '/' not allowed", name)
```
Comment (singholt) [inline ecs-init/exec/iptables/iptables.go:234]: nit: ~S~space characters

File: ecs-init/exec/iptables/iptables.go (hunk 1 above)
Comment (sparrc) [reply ecs-init/exec/iptables/iptables.go]: done

File: ecs-init/exec/iptables/iptables.go (hunk 2 above)
Comment (sparrc) [reply ecs-init/exec/iptables/iptables.go]: reworded

**Review by singholt — APPROVED** [review]

**Review by danehlim — APPROVED** [review]

**Pull request merged**